src/main/webapp/scripts/map.js

```diff
@@ -154,31 +163,58 @@ function getColourMarker(category) {
 }
 
 /* Place Marker Where Map is Clicked On & Show Popup*/
-function placeMarkerAndPanTo(latLng, map) {
-  var marker = new google.maps.Marker({
+function placeMarkerAndPanTo(latLng) {
+  /* Make marker for the clicked location */
+  const marker = new google.maps.Marker({
     position: latLng,
     map: map,
     icon: greyIcon
   });
-
+  /* Recenter the Map */
   map.panTo(latLng);
-  togglePopup();
+  /* Show the popup box */
+  togglePopup(latLng);
+  /* Update the latitude and longitude values in the popup */
   var markerPosition = marker.getPosition();
   populateLocation(markerPosition);
 
   /* If marker is right-clicked, delete */
-  google.maps.event.addListener(marker, 'rightclick', function(event) {
-      marker.setMap(null);
+  google.maps.event.addListener(marker, "rightclick", function(event) {
+    marker.setMap(null);
   });
 }
 
 /* Set the PopUp to Active */
-function togglePopup() {
+function togglePopup(latLng) {
   document.getElementById("popup-add-recs").classList.toggle("active");
+  document.getElementById("submit-recommendation").addEventListener("click", () =>
+    postMarker(latLng), false);
```

Comment:
We need to redesign this part I think. If the user puts two markers, the `submit-recommendation` div will have two event listeners. When the user will perform the click action after putting the second marker (and discarding the first marker), `postMarker` will be called two times with the first `latLng` and the second `latLong`. 

I tried to put a simple demo [here](http://screen/7AfYDmBFViX5JtB). 

I might be wrong, could you please double check with some console prints maybe?

Reply:
Thanks for the meeting to go through this and help me to understand the problem!